cpp/tests/sampling/mg_negative_sampling.cpp

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utilities/base_fixture.hpp"
+#include "utilities/conversion_utilities.hpp"
+#include "utilities/property_generator_utilities.hpp"
+#include "utilities/validation_utilities.hpp"
+
+#include <cugraph/sampling_functions.hpp>
+#include <cugraph/utilities/high_res_timer.hpp>
+
+#include <gtest/gtest.h>
+
+struct Negative_Sampling_Usecase {
+  float sample_multiplier{2};
+  bool use_src_bias{false};
+  bool use_dst_bias{false};
+  bool remove_duplicates{false};
+  bool remove_existing_edges{false};
+  bool exact_number_of_samples{false};
```

Comment:
What would behavior be in an edge masked graph?  That if an edge is masked out it would be allowed to be in the negative sampled edges if `remove_existing_edges` is set to true?  I can add that tests for that.

Reply:
Yes. If you call has_edge() on masked out edges, you will get false. So, those edges can be included in the negative sampling output

Reply:
Done in next push